packages/rest/src/__tests__/unit/rest.server/rest.server.open-api-spec.unit.ts

```diff
@@ -375,6 +386,26 @@ describe('RestServer.getApiSpec()', () => {
     expect(spec.info).to.eql(EXPECTED_SPEC_INFO);
   });
 
+  context('OpenApiSpecOptions', () => {
+    it('disables consolidator if disableOASConsolidator is set to true', async () => {
+      const options: {rest: RestServerConfig} = {
+        rest: {openApiSpec: {disableOASConsolidator: true}},
+      };
+      options.rest = givenHttpServerConfig(options.rest);
+      app = new Application(options);
+      app.component(RestComponent);
+      server = await app.getServer(RestServer);
+      await server.start();
+
+      const enhancer = (await server.OASEnhancer.getEnhancerByName(
+        'consolidate',
+      )) as ConsolidationEnhancer;
```

Comment:
I consider explicit code cast as a code smell, can we find a way how to avoid it here?

For example, we can improve `getEnhancerByName` to accept a generic parameter.

```suggestion
    const enhancer = await server.OASEnhancer.getEnhancerByName<
      ConsolidationEnhancer
    >('consolidate');
```

Implementation in `spec-enhancer.service.ts`:

```ts
export class OASEnhancerService {
  // ...

  async getEnhancerByName<T extends OASEnhancer>(
    name: string,
  ): Promise<T | undefined> {
    // Get the latest list of enhancers
    const enhancers = await this.getEnhancers();
    return enhancers.find(e => e.name === name) as T | undefined;
  }
}
```

Thoughts?

Reply:
As always much better with your direction. 👍 I've made the change to getEnhancerByName but couldn't get test to work so fell back to running schema through consolidator.

Reply:
Maybe the problem was caused by missing default value for the new generic argument?

```ts
export class OASEnhancerService {
  // ...

  async getEnhancerByName<T extends OASEnhancer = OASEnhancer >(
  // ...
```

Reply:
I think the issue is that I was grabbing the class when I wanted an instance, so it wouldn't have worked.  I have left the change (excluding default) to getEnhancerByName in as a commit.  Squashed commits into one ready for merge.